src/ngraph/runtime/gpu/gpu_runtime_context.hpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include "ngraph/runtime/gpu/gpu_cuda_context_manager.hpp"
+#include "ngraph/runtime/gpu/gpu_cuda_function_pool.hpp"
+#include "ngraph/runtime/gpu/gpu_util.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace gpu
+        {
+            extern "C" {
```

Comment:
extern "C" here seems wrong on many levels. namespaces are C++ and the ctor/dtor in the struct are C++. I don't understand this.

Reply:
@rkimballn1, this was a mistake in overall design of the GPURuntimeContext, which I have corrected in the most recent commit. 

For the GPURuntimeContext struct it is quite important that it be extern "C" so that we are only using the C API when operating on ngraph data in the JIT compiled function. This point was raised to me by @jmenon and @jbobba, where the local clang implementation may target a different GLIBC ABI than the native compiler that was used to build nGraph. In these cases, calling nGraph C++ from within the clang could cause runtime errors. By forcing the interface to be C-only, this problem is avoided.